decidim-proposals/spec/types/withdraw_proposal_type_spec.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "decidim/api/test/mutation_context"
+
+module Decidim
+  module Proposals
+    describe WithdrawProposalType, type: :graphql do
+      include_context "with a graphql class mutation"
+
+      let(:root_klass) { ProposalMutationType }
+      let(:organization) { create(:organization, available_locales: [:en]) }
+      let(:participatory_process) { create(:participatory_process, :with_steps, organization:) }
+      let(:proposal_component) { create(:proposal_component, participatory_space: participatory_process) }
+      let(:author) { create(:user, :confirmed, organization:) }
+      let!(:model) { create(:proposal, component: proposal_component, users: [author]) }
+      let(:component) { model.component }
+      let(:query) do
+        <<~GRAPHQL
+          mutation() {
+            withdraw(input: {}) {
+              id
+              state
+              withdrawnAt
+            }
+          }
+        GRAPHQL
+      end
+
+      let(:variables) do
+        {
+          input: {
+            attributes: {}
+          }
+        }
+      end
+
+      context "when withdrawing a proposal" do
```

Comment:
I think semantically a describe is better instead of a context. To better support my argument, you're not redefining or refining the context below of the definition (i.e. it doesn't have any variable just below this line)

```suggestion
      describe "withdrawing a proposal" do
```